hibernate-core/src/main/java/org/hibernate/hql/spi/id/cte/CteValuesListDeleteHandlerImpl.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.hql.spi.id.cte;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hibernate.engine.spi.QueryParameters;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.hql.internal.ast.HqlSqlWalker;
+import org.hibernate.hql.spi.id.MultiTableBulkIdStrategy;
+import org.hibernate.persister.collection.AbstractCollectionPersister;
+import org.hibernate.sql.Delete;
+import org.hibernate.type.CollectionType;
+import org.hibernate.type.Type;
+
+/**
+ * Bulk-id delete handler that uses CTE and VALUES lists.
+ *
+ * @author Evandro Pires da Silva
+ * @author Vlad Mihalcea
+ */
+public class CteValuesListDeleteHandlerImpl
+		extends AbstractCteValuesListBulkIdHandler
+		implements MultiTableBulkIdStrategy.DeleteHandler {
+
+	private final List<String> deletes = new ArrayList<>();
+
+	public CteValuesListDeleteHandlerImpl(
+			SessionFactoryImplementor factory,
+			HqlSqlWalker walker) {
+		this( factory, walker, null, null );
+	}
+
+	public CteValuesListDeleteHandlerImpl(
+			SessionFactoryImplementor factory,
+			HqlSqlWalker walker,
+			String catalog,
+			String schema) {
+		super( factory, walker, catalog, schema );
+
+		final String idSubselect = generateIdSubselect( getTargetedQueryable() );
+
+		// If many-to-many, delete the FK row in the collection table.
+		// This partially overlaps with DeleteExecutor, but it instead uses the
+		// temp table in the idSubselect.
+		for ( Type type : getTargetedQueryable().getPropertyTypes() ) {
+			if ( type.isCollectionType() ) {
+				CollectionType cType = (CollectionType) type;
+				AbstractCollectionPersister cPersister = (AbstractCollectionPersister) factory.getCollectionPersister( cType.getRole() );
```

Comment:
`factory.getCollectionPersister(...)` is deprecated.
Same in `AbstractInlineIdsDeleteHandlerImpl`